Compilation error during install

Hi, first off thank you for this work, it looks very useful.
I'm trying to get the torch-geometry package to work but torch-cluster fails to compile.

```
$ pipenv run python --version
Python 3.7.0b2

$ pipenv run python -c "import torch; print(torch.__version__)"
0.4.1.post2

$ nvcc --version
nvcc: NVIDIA (R) Cuda compiler driver
Copyright (c) 2005-2018 NVIDIA Corporation
Built on Tue_Jun_12_23:07:04_CDT_2018
Cuda compilation tools, release 9.2, V9.2.148
```

I get following error message:
```
$ pipenv install torch-cluster
Installing torch-cluster...
Collecting torch-cluster
  Using cached https://files.pythonhosted.org/packages/36/b0/25ca8b811059e001f1e3285ac3036b6969fb21350e411c6881ba2b9be3c3/torch_cluster-1.1.2.tar.gz
Requirement already satisfied: cffi in /home/philipp/dotfiles/stow/t460p/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages (from torch-cluster) (1.11.5)
Requirement already satisfied: pycparser in /home/philipp/dotfiles/stow/t460p/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages (from cffi->torch-cluster) (2.18)
Building wheels for collected packages: torch-cluster
  Running setup.py bdist_wheel for torch-cluster: started
  Running setup.py bdist_wheel for torch-cluster: finished with status 'error'
  Complete output from command /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/bin/python3.7m -u -c "import setuptools, tokenize;__file__='/tmp/pip-install-p1cl3unl/torch-cluster/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" bdist_wheel -d /tmp/pip-wheel-exs4dzw9 --python-tag cp37:
  Compiling kernel...
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.7
  creating build/lib.linux-x86_64-3.7/torch_cluster
  copying torch_cluster/graclus.py -> build/lib.linux-x86_64-3.7/torch_cluster
  copying torch_cluster/__init__.py -> build/lib.linux-x86_64-3.7/torch_cluster
  copying torch_cluster/grid.py -> build/lib.linux-x86_64-3.7/torch_cluster
  creating build/lib.linux-x86_64-3.7/test
  copying test/test_graclus.py -> build/lib.linux-x86_64-3.7/test
  copying test/utils.py -> build/lib.linux-x86_64-3.7/test
  copying test/__init__.py -> build/lib.linux-x86_64-3.7/test
  copying test/test_grid.py -> build/lib.linux-x86_64-3.7/test
  creating build/lib.linux-x86_64-3.7/torch_cluster/_ext
  copying torch_cluster/_ext/__init__.py -> build/lib.linux-x86_64-3.7/torch_cluster/_ext
  creating build/lib.linux-x86_64-3.7/torch_cluster/utils
  copying torch_cluster/utils/ffi.py -> build/lib.linux-x86_64-3.7/torch_cluster/utils
  copying torch_cluster/utils/loop.py -> build/lib.linux-x86_64-3.7/torch_cluster/utils
  copying torch_cluster/utils/perm.py -> build/lib.linux-x86_64-3.7/torch_cluster/utils
  copying torch_cluster/utils/__init__.py -> build/lib.linux-x86_64-3.7/torch_cluster/utils
  creating build/lib.linux-x86_64-3.7/torch_cluster/_ext/ffi
  copying torch_cluster/_ext/ffi/__init__.py -> build/lib.linux-x86_64-3.7/torch_cluster/_ext/ffi
  warning: build_py: byte-compiling is disabled, skipping.
  
  running build_ext
  generating cffi module 'build/temp.linux-x86_64-3.7/torch_cluster._ext.ffi._ffi.c'
  creating build/temp.linux-x86_64-3.7
  building 'torch_cluster._ext.ffi._ffi' extension
  creating build/temp.linux-x86_64-3.7/build
  creating build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7
  creating build/temp.linux-x86_64-3.7/tmp
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl/torch-cluster
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl/torch-cluster/aten
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl/torch-cluster/aten/TH
  creating build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl/torch-cluster/aten/THCC
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/THC -Iaten/TH -Iaten/THCC -I/home/philipp/.pyenv/versions/3.7.0b2/include/python3.7m -c build/temp.linux-x86_64-3.7/torch_cluster._ext.ffi._ffi.c -o build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7/torch_cluster._ext.ffi._ffi.o -std=c99 -std=c99
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/THC -Iaten/TH -Iaten/THCC -I/home/philipp/.pyenv/versions/3.7.0b2/include/python3.7m -c /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGraclus.c -o build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGraclus.o -std=c99 -std=c99
  gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/THC -Iaten/TH -Iaten/THCC -I/home/philipp/.pyenv/versions/3.7.0b2/include/python3.7m -c /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c -o build/temp.linux-x86_64-3.7/tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.o -std=c99 -std=c99
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateFloatTypes.h:10,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:10,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THFloatTensor_grid’:
  aten/TH/generic/THGrid.c:15:49: error: dereferencing pointer to incomplete type ‘THTensor {aka struct THTensor}’
         value += coef * (int64_t) (posData[d * pos->stride[1]] / sizeData[d]);
                                                   ^~
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateFloatTypes.h:11,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:10,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THDoubleTensor_grid’:
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:10,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THByteTensor_grid’:
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:11,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THCharTensor_grid’:
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:12,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THShortTensor_grid’:
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:13,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THIntTensor_grid’:
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  In file included from generic/THGrid.c:1:0,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:14,
                   from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                   from /tmp/pip-install-p1cl3unl/torch-cluster/aten/TH/THGrid.c:4:
  aten/TH/generic/THGrid.c: In function ‘THLongTensor_grid’:
  aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
     real *posData = THTensor_(data)(pos);
           ^~~~~~~
  error: command 'gcc' failed with exit status 1
```

### I also tried to install from source, with pretty much the same error:

```
$ pipenv install --dev -e ~/tmp/pytorch_cluster
Installing -e /home/philipp/tmp/pytorch_cluster...
Obtaining file:///home/philipp/tmp/pytorch_cluster
Requirement already satisfied: cffi in /home/philipp/dotfiles/stow/t460p/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages (from torch-cluster==1.1.2) (1.11.5)
Requirement already satisfied: pycparser in /home/philipp/dotfiles/stow/t460p/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages (from cffi->torch-cluster==1.1.2) (2.18)
Installing collected packages: torch-cluster
  Running setup.py develop for torch-cluster
    Complete output from command /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/bin/python3.7m -c "import setuptools, tokenize;__file__='/home/philipp/tmp/pytorch_cluster/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" develop --no-deps:
    Compiling kernel...
    running develop
    running egg_info
    writing torch_cluster.egg-info/PKG-INFO
    writing dependency_links to torch_cluster.egg-info/dependency_links.txt
    writing requirements to torch_cluster.egg-info/requires.txt
    writing top-level names to torch_cluster.egg-info/top_level.txt
    reading manifest file 'torch_cluster.egg-info/SOURCES.txt'
    reading manifest template 'MANIFEST.in'
    writing manifest file 'torch_cluster.egg-info/SOURCES.txt'
    running build_ext
    generating cffi module 'build/temp.linux-x86_64-3.7/torch_cluster._ext.ffi._ffi.c'
    creating build/temp.linux-x86_64-3.7
    building 'torch_cluster._ext.ffi._ffi' extension
    creating build/temp.linux-x86_64-3.7/build
    creating build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7
    creating build/temp.linux-x86_64-3.7/home
    creating build/temp.linux-x86_64-3.7/home/philipp
    creating build/temp.linux-x86_64-3.7/home/philipp/tmp
    creating build/temp.linux-x86_64-3.7/home/philipp/tmp/pytorch_cluster
    creating build/temp.linux-x86_64-3.7/home/philipp/tmp/pytorch_cluster/aten
    creating build/temp.linux-x86_64-3.7/home/philipp/tmp/pytorch_cluster/aten/TH
    creating build/temp.linux-x86_64-3.7/home/philipp/tmp/pytorch_cluster/aten/THCC
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/THC -Iaten/TH -Iaten/THCC -I/home/philipp/.pyenv/versions/3.7.0b2/include/python3.7m -c build/temp.linux-x86_64-3.7/torch_cluster._ext.ffi._ffi.c -o build/temp.linux-x86_64-3.7/build/temp.linux-x86_64-3.7/torch_cluster._ext.ffi._ffi.o -std=c99 -std=c99
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/THC -Iaten/TH -Iaten/THCC -I/home/philipp/.pyenv/versions/3.7.0b2/include/python3.7m -c /home/philipp/tmp/pytorch_cluster/aten/TH/THGraclus.c -o build/temp.linux-x86_64-3.7/home/philipp/tmp/pytorch_cluster/aten/TH/THGraclus.o -std=c99 -std=c99
    gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -fPIC -DWITH_CUDA -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH -I/home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/THC -Iaten/TH -Iaten/THCC -I/home/philipp/.pyenv/versions/3.7.0b2/include/python3.7m -c /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c -o build/temp.linux-x86_64-3.7/home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.o -std=c99 -std=c99
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateFloatTypes.h:10,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:10,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THFloatTensor_grid’:
    aten/TH/generic/THGrid.c:15:49: error: dereferencing pointer to incomplete type ‘THTensor {aka struct THTensor}’
           value += coef * (int64_t) (posData[d * pos->stride[1]] / sizeData[d]);
                                                     ^~
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateFloatTypes.h:11,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:10,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THDoubleTensor_grid’:
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:10,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THByteTensor_grid’:
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:11,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THCharTensor_grid’:
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:12,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THShortTensor_grid’:
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:13,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THIntTensor_grid’:
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    In file included from generic/THGrid.c:1:0,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateIntTypes.h:14,
                     from /home/philipp/.local/share/virtualenvs/pytorch_geometric-4vUoAfdD/lib/python3.7/site-packages/torch/utils/ffi/../../lib/include/TH/THGenerateAllTypes.h:11,
                     from /home/philipp/tmp/pytorch_cluster/aten/TH/THGrid.c:4:
    aten/TH/generic/THGrid.c: In function ‘THLongTensor_grid’:
    aten/TH/generic/THGrid.c:7:9: warning: variable ‘posData’ set but not used [-Wunused-but-set-variable]
       real *posData = THTensor_(data)(pos);
             ^~~~~~~
    error: command 'gcc' failed with exit status 1
```

I got all other dependencies of torch-geometry to work but I'm failing here.
Would appreciate your support.

Also torch-spline-conv only installed from source but not from pypi, where I get a similar compilation error. I can add another issue if needed.

--- a/aten/TH/generic/THGrid.c
+++ b/aten/TH/generic/THGrid.c
@@ -6,16 +6,18 @@ void THTensor_(grid)(THLongTensor *self, THTensor *pos, THTensor *size, THLongTe
   int64_t *selfData = THLongTensor_data(self);
   real *posData = THTensor_(data)(pos);
   real *sizeData = THTensor_(data)(size);
+  int64_t posStride0 = THTensor_(stride)(pos, 0);
+  int64_t posStride1 = THTensor_(stride)(pos, 1);
   int64_t *countData = THLongTensor_data(count);
 
   ptrdiff_t n, d; int64_t coef, value;
   for (n = 0; n < THTensor_(size)(pos, 0); n++) {
     coef = 1; value = 0;
     for (d = 0; d < THTensor_(size)(pos, 1); d++) {
-      value += coef * (int64_t) (posData[d * pos->stride[1]] / sizeData[d]);
+      value += coef * (int64_t) (posData[d * posStride1] / sizeData[d]);
       coef *= countData[d];
     }
-    posData += pos->stride[0];
+    posData += posStride0;
     selfData[n] = value;
   }
 }

